src/goto-programs/remove_function_pointers.cpp

```diff
@@ -362,40 +367,55 @@ void remove_function_pointerst::remove_function_pointer(
   assert(function.operands().size()==1);
 
   const exprt &pointer=function.op0();
+  remove_const_function_pointerst::functionst functions;
+  remove_const_function_pointerst fpr(
+    *message_handler, pointer, ns, symbol_table);
 
-  // Is this simple?
-  if(pointer.id()==ID_address_of &&
-     to_address_of_expr(pointer).object().id()==ID_symbol)
+  bool found_functions=fpr(functions);
+
+  // Consistency checks
+  if(found_functions && functions.size()==0)
```

Comment:
For [`std::set`](http://en.cppreference.com/w/cpp/container/set/size)/[`std::unordered_set`](http://en.cppreference.com/w/cpp/container/unordered_set/size) (which is was `functions` is) `size()` is constant time. None the less, I think `empty()` expresses the check clearer so I will do this.